packages/valory/skills/decision_maker_abci/behaviours/storage_manager.py

```diff
@@ -253,19 +270,111 @@ def _fetch_accuracy_info(self) -> Generator[None, None, bool]:
                 f"Could not retrieve data from the url {accuracy_link}. "
                 f"Received status code {response.status_code}."
             )
-            return False
+            return None
 
-        self.context.logger.info("Parsing accuracy information of the tools...")
         try:
-            self.accuracy_information = StringIO(response.body.decode())
+            tool_accuracy_data = StringIO(response.body.decode())
+            self.context.logger.info("Parsing accuracy information of the tools...")
+            return tool_accuracy_data
+
         except (ValueError, TypeError) as e:
             self.context.logger.error(
                 f"Could not parse response from ipfs server, "
                 f"the following error was encountered {type(e).__name__}: {e}"
             )
-            return False
+            return None
 
-        return True
+    def _set_accuracy_info_from_remote(
+        self,
+            remote_accuracy_info: Generator[None, None, Optional[StringIO]]
+    ) -> None:
+        """Set the tool accuracy info from the remote store."""
+        tool_accuracy_data = remote_accuracy_info
+
+        if tool_accuracy_data is not None:
+            self.accuracy_information = tool_accuracy_data
+
+    def _read_local_policy_date(self) -> int:
+        """Reads the updated timestamp from the policy file."""
```

Comment:
This method is too hacky too, I do not think it is necessary if we update the decoder.